nq/cli.py

```diff
@@ -115,6 +123,15 @@ def main():
         "--message",
         help="Commit message for the main repo (default: 'Update [repo] to latest')",
     )
+    pull_parser.add_argument(
+        "--commit-sha",
```

Comment:
```suggestion
        "--ref",
```
We can generalize this further with little change